q2_composition/_ancombc.py

```diff
@@ -139,10 +139,19 @@ def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
     for level in reference_levels:
         try:
             column, level_value = level.split('::')
-        except Exception:
-            raise ValueError('Too many column-value pair separators found'
+        except ValueError as e:
+            # More than one ::
+            if 'too many' in str(e):
```

Comment:
Parsing Python's error messages feels risky to me as we don't have any guarantee (that I'm aware of) that those will be stable across versions. I would more explicitly check for the problems, something like the following (this code hasn't been tested so probably doesn't work as-is):

```python
fields = level.split('::', maxsplit=1)
if len(fields) == 1: 
    # raise your "too few" error

column, level_value = fields

if column not in md.columns:
    message = 'Column %s not in metadata.'
    if ':' in column:
        message = f"{message} This column name appears to contain a `:`, which can be a problem for this action."
...
```

Reply:
I think you're right that they could just decide to change those error messages. I've addressed this and probably done enough to provide a reasonable stop gap until we can address the issue using a collection. Maybe @lizgehret and I can do that when this library stuff is done and she can start doing the Collections in usage examples and galaxy and all that.